pandas/core/internals/managers.py

```diff
@@ -350,25 +351,62 @@ def _verify_integrity(self) -> None:
                 f"tot_items: {tot_items}"
             )
 
-    def reduce(self, func, *args, **kwargs):
+    def reduce(self, func, ignore_failures=False):
         # If 2D, we assume that we're operating column-wise
         if self.ndim == 1:
             # we'll be returning a scalar
             blk = self.blocks[0]
-            return func(blk.values, *args, **kwargs)
+            return func(blk.values)
 
         res = {}
         for blk in self.blocks:
-            bres = func(blk.values, *args, **kwargs)
+            placement = blk.mgr_locs.as_array
+            if isinstance(blk, CategoricalBlock):
+                try:
+                    bres = func(blk.values)
+                except TypeError:
+                    # not all operations (eg any, all) are supported on
+                    # Categorical, so fallback to operating on dense array
+                    # eg pandas/tests/frame/test_analytics.py::TestDataFrameAnalytics::test_any_all_np_func
+                    bres = func(np.asarray(blk.values).reshape(1, len(blk.values)))
+            elif isinstance(blk, ObjectBlock):
```

Comment:
This long block is to handle object dtype column-wise when it fails on all values (since it object can be mixed). Could also go directly to column-wise always avoiding the first try/except.